roofit/roofitcore/src/RooProdPdf.cxx

```diff
@@ -1189,9 +1184,7 @@ void RooProdPdf::rearrangeProduct(RooProdPdf::CacheElem& cache) const
 
 //    cout << "denominator components are " << dentmp->components() << endl ;
    RooArgSet comps(dentmp->components()) ;
-   RooFIter piter = comps.fwdIterator() ;
-   RooAbsReal* parg ;
-   while((parg=(RooAbsReal*)piter.next())) {
+   for (auto* parg : static_range_cast<RooAbsReal*>(comps)) {
```

Comment:
```suggestion
   for (auto* parg : static_range_cast<RooAbsReal*>(dentmp->components())) {
```

I don't this that `comps` is used anywhere else, so you can delete the precious line and avoid creating a RooArgSet by copy here I think.